ntrcard.cpp

```diff
@@ -180,7 +183,16 @@ bool sendCommand(const uint64_t cmd, uint16_t response_len, uint8_t *resp, OpFla
 }
 
 bool init() {
-    platform::resetCard();
+    if (platform::CAN_RESET) {
+        platform::resetCard();
+        state.status = Status::RAW;
+    } else if (state.status > Status::RAW) {
+        platform::logMessage(LOG_ERR,
+            "Trying to re-init after encryption is on on !CAN_RESET platform"
+            " (status = %d)",
+            static_cast<uint32_t>(state.status));
+        return false;
+    }
```

Comment:
Sorry, I cannot remember, how can send DUMMY command on the other platform?

Reply:
`ntrcard::init` does that.

Reply:
KEY2 has a lower value than RAW? I think will return with false.

Reply:
`KEY2 > RAW`.

[See what the compiler says.](http://cpp.sh/85eg)

Reply:
Or do you mean how to send 0x9F while in KEY2?

As I said before that is nonsense and I don't believe any cart requires 0x9F _as in the NTR card initialisation command_ as part of the initialisation.

If it does, that is a _cart-specific_ initialisation command and is distinct from the 0x9F sent immediately after reset, so the flashcart class should just use `sendCommand` to send that.

Reply:
are you have ak2i? i just said with my memory (so I can be wrong)
in my test that(send 9f) needs for detection card & chip id
(also i don't remember flag something)

so if you have, please test it. and if not, we should remain same process until check.

Reply:
> we should remain same process until check.

Again, if 0x9F is required, it should be sent by the AK2i class.

Not here.

Reply:
1. please remove compiler comment. it's reasonable reviews. it looks judge my skill.
2. it's not a sense or nonsense problem, just feature change/unchange problem at the refactor.
  you want to change reset part on the ds/dsi section, but this code can't
  so we need to result check from all other carts.

Reply:
how about DSTT.
I mean, if you change something and that can change logic process, you should also fix that.

Reply:
> please remove compiler comment. it's reasonable reviews. it looks judge my skill.

No one is doubting you or your skill. I wanted to be sure myself, so I went to check and then linked the snippet here.

But OK, I'll remove it.

> it's not a sense or nonsense problem, just feature change/unchange problem at the refactor.

If the AK2i requires 0x9F to access flashcart commands in KEY2, then it should be done in the AK2i class.

The `init` function sends [the dummy command](http://problemkaputt.de/gbatek.htm#dscartridgeprotocol) as part of the NTR card initialisation process _in RAW_. 0x9F in KEY2 is _not_ the same thing and `init` is _definitely_ the wrong place for it.

Reply:
1. okay. thanks.

2. I remember ak2i cannot detect chip and cart id without SEC flags. so cannot set RAW at that time.
ak2i uses second section and flashcart feature alive after install ntrboot exploit. maybe it's magic, but needs also other carts.

3. I agree as your words(here is wrong place, and should add to ak2i).
  but I said it's not a sense/nonsense problem. I said risk of logic changes at the refactoring something without test.
  well... if just add this code to master (without any more changes), I'll send 9f command at the each cart initialization...